### ISSUE TITLE
URGENT BLOCKER: Dialog list not updating

### ISSUE THREAD

**Issue by hagitmer** [Hacknocraft/AppFriendsAndroidCore#14]
Unread badges are not updating, neither are latest messages.  It looks like it is due to a non-fatal crash coming from unhandled backpressure on an observable.

**Comment by nrschultz** [comment]
Here's the stacktrace, the "Error fetching dialogs" message is ours, but it seems like that is just happening because its our code at the end of the subscriber chain.  

```
E/messaging: Error fetching dialogs
             rx.exceptions.MissingBackpressureException
                 at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.onNext(OperatorObserveOn.java:160)
                 at rx.internal.operators.OperatorSubscribeOn$1$1.onNext(OperatorSubscribeOn.java:53)
                 at rx.internal.operators.NotificationLite.accept(NotificationLite.java:135)
                 at rx.subjects.SubjectSubscriptionManager$SubjectObserver.emitNext(SubjectSubscriptionManager.java:253)
                 at rx.subjects.BehaviorSubject.onNext(BehaviorSubject.java:160)
                 at me.appfriends.sdk.dialog.n$1.a(Unknown Source)
                 at me.appfriends.sdk.dialog.n$1.onNext(Unknown Source)
                 at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:134)
                 at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.call(OperatorObserveOn.java:224)
                 at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:228)
                 at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
                 at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
                 at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                 at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:269)
                 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                 at java.lang.Thread.run(Thread.java:818)
```

**Comment by hagitmer** [comment]
@laeroah can this be included in tomorrow's release? It's a major blocker for us at this point. Thanks

**Comment by laeroah** [comment]
We have fixed it and doing QA internally. Will release to you soon.

**Comment by laeroah** [comment]
We have fixed this, and will release to you soon, and more optimizations will follow.

**Comment by laeroah** [comment]
should be fixed by 3.0.6 release

**Comment by HacknocraftDev** [comment]
This issue was moved to Hacknocraft/AppFriendsAndroidCore#15
